### ISSUE TITLE
Three Tier Administration

### ISSUE THREAD

**Issue by tpnsolutions** [humhub/humhub#2833]
I'd like to setup a "three-tier" admin schema.

Super Admin
who has access to every feature, and basically can do anything.

Regular Admin
who has access to whatever the Super Admin gives them access to, but cannot delete the Super Admin.

Moderator
who has access to approving content, and potentially new User registrations, but cannot delete either a Regular or Super Admin nor can they tweak settings.

Presently during my testing, anyone I give access to "Manage Users" or "Manage Groups" can delete another Admin, which is not ideal. At the very least, there should be "one" Admin who is granted all permissions, and cannot be deleted, however in practice things don't seem to be setup this way presently.

Any help would be greatly appreciated :-)

Cheers!

**Comment by Felli** [comment]
I can agree that the permissions system isn't 100% as of yet, and agree that these solutions could benefit HumHub for the better, although I must suggest that this is moved over to the [community site](https://community.humhub.com) under the Feature Request space so that we can better log any and all requests better. :+1:

**Comment by tpnsolutions** [comment]
Fair enough. I've posted the idea in the community.

I must however ask, if logging an issue in an "issue tracker" results in being forwarded to another site, what purpose does the "issue tracker" have?

From an "issue" point of view, being able to delete a "super admin" (someone with more permissions than you) shouldn't be possible. Point being is, I created a group for "regular admins" and disabled features I didn't want them having access to. However, giving them access to manage users, gave them the ability to delete the "super admin" who had the full permission set. This meant that you could no longer access these additional features, without manually adjusting the database and either giving a "regular admin", "super admin" privileges, or re-creating the missing "super admin" profile.

Long story short, while my original post outlined a "3-tier" structure, the core issue above is what prompted this discussion in the first place. Hence, while I've got no problem posting the bigger idea in the "community", I do believe the above noted "issue" does deserve some sort of attention as I've never run into a program which actively let a "less privileged" admin, wipe out the "super admin". This simply isn't logical.

**Comment by Felli** [comment]
The GitHub repository issue tracker is for tracking bugs and other issues with the platform, general enhancements and such are normally preferred to go through the community site which always has a staff member active at different times, I can agree that we lack a lot of permissions as I stated in my comment above, as your suggestions would indeed really help with this and enhance the permissions that are in place at this time, I myself have suggested in the past that global permissions should be enhanced for better use, which has already started growing. :smile:

**Comment by buddh4** [comment]
You can assign global group permissions as `Manage Users`, `Manage Spaces`, `Manage Modules`, etc. which allows non system admins to access some of the admin sections. In the future we'll have to spread those permissions also on space level functionality.

**Comment by tpnsolutions** [comment]
The problem with the out of the box install is that anyone with Manage Users permission can delete any other user including an admin with more permissions. This can cause functionality to become orphaned.

In my opinion, there should be a super admin. This access level should be the only level able to delete an admin.

**Comment by buddh4** [comment]
This just has to be added to the delete user functionality, I think I've added this to the 1.3 todo list, since there was a related ticket.

The Admin group alsready is marked as super admin group, and you can check if the current user is part of this group by `Yii::$app->user->getIdentity()->isSystemAdmin()`.

I think you should also be able to implement a [BaseAdminPermission](https://github.com/humhub/humhub/blob/master/protected/humhub/modules/admin/components/BaseAdminPermission.php#L19) without exporting it, so the state should not be changable. Unfortunately there is no `isFixedGroup($goupId)` function to prevent the change of other group states.
